File: Algorithm/Go/1000+/1566. Detect Pattern of Length M Repeated K or More Times.go
func containsPattern(arr []int, m int, k int) bool {
	for i := 0; i < len(arr)-m*k+1; i++ {
		for j := 0; j < k; j++ {
			if !equal(arr[i:i+m], arr[i+m*j:i+m*j+m]) {
				break
			}
			if j == k-1 {
				return true
			}
		}
	}
	return false
}

func equal(a, b []int) bool {
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}

File: Algorithm/C++/1000+/1566. Detect Pattern of Length M Repeated K or More Times.cc
#include <vector>
#include <algorithm>

class Solution {
public:
    bool containsPattern(std::vector<int>& arr, int m, int k) {
        for (auto start=arr.begin(); start < arr.end() - m*k + 1; start++) {
            // std::vector<int> sub(start, start+m); //slice vecor
            auto tmp = start;
            for(int i=0; i < k; i++) {
                if (!std::equal(start, start+m, tmp)) 
                    break;
                if (i==k-1) 
                    return true;
                tmp += m;
            }
        }        
        return false;
    }
};